docs/current_docs/manuals/developer/python/820256-module-structure.mdx

```diff
@@ -5,36 +5,81 @@ toc_max_heading_level: 2
 title: "Module Structure"
 ---
 
-
 # Module Structure
 
-
 It's essential to understand a few key concepts about Dagger Modules created for use with the Python SDK, for a better fit with your normal development workflow.
 
-## Runtime container with Python 3.11
+## Python version
+
+Dagger Modules run in a [container](../overview/942201-execution-environment.mdx) that's bootstrapped specifically by the Python SDK, with the necessary environment to run the module's code. It currently runs Python 3.11 by default but starting with [v0.11.0][v11] that can be changed.
+
+The desired interpreter version can be pinned by creating a `.python-version` file next to your `pyproject.toml`:
+```console
+echo "3.11" > .python-version
+```
+
+This can be useful either to pin the version in case the default one gets an upgrade, or to choose another a newer version sooner.
+
+As an alternative, setting [requires-python](https://packaging.python.org/en/latest/guides/writing-pyproject-toml/#python-requires) in `pyproject.toml` will also pin the version, but using a lower bound constraint (minimum version). For example, the following will use 3.11, even though there's a 3.12 available:
+
+```toml
+[project]
+requires-python = ">=3.11"
+```
+
+## Base image
 
-Dagger Modules run in a container (called the "runtime container") that's bootstrapped specifically by the Python SDK, with the necessary environment to run the module's code. It's currently hardcoded to run in Python 3.11 (although this may be configurable in future).
+The default base image is [python:VERSION-slim](https://hub.docker.com/_/python/tags?name=3.11-slim) (based on the [Python version](#python-version)), but it can be overridden in `pyproject.toml` since [v0.11][v11]:
```

Comment:
```suggestion
The default base image is [python:VERSION-slim](https://hub.docker.com/_/python/tags?name=3.11-slim) (based on the [Python version](#python-version)), but it can be overridden in `pyproject.toml` since [Dagger v0.11][v11]:
```